packages/firebase_crashlytics/lib/src/firebase_crashlytics.dart

```diff
@@ -0,0 +1,185 @@
+// Copyright 2019, the Flutter project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+part of firebase_crashlytics;
+
+/// The entry point for accessing Crashlytics.
+///
+/// You can get an instance by calling `Crashlytics.instance`.
+class Crashlytics {
+  static final Crashlytics instance = Crashlytics();
+
+  /// Set to true to have Errors sent to Crashlytics while in debug mode.
+  bool reportInDevMode = false;
+
+  /// Keys to be included with report.
+  @visibleForTesting
+  final Map<String, dynamic> keys = <String, dynamic>{};
+
+  /// Logs to be included with report.
+  @visibleForTesting
+  final ListQueue<String> logs = ListQueue<String>(15);
+  int logSize = 0;
+
+  bool get isInDebugMode {
+    bool _inDebugMode = false;
+    if (!reportInDevMode) {
+      assert(_inDebugMode = true);
+    }
+    return _inDebugMode;
+  }
+
+  static const MethodChannel channel =
+      MethodChannel('plugins.flutter.io/firebase_crashlytics');
+
+  Future<void> onError(FlutterErrorDetails details) async {
+    print('Error caught by Crashlytics plugin:');
+    if (isInDebugMode && !reportInDevMode) {
+      print(Trace.format(details.stack).trimRight().split('\n'));
+    } else {
+      // Send logs
+      await sendLogs();
```

Comment:
Can you merge the sending of logs into the `onError` call below? I'm not sure it needs its own method call.

Reply:
Done

Reply:
sorry to add my two cents here again.. but imho the `sendLogs()` method was a nice way for me as a user of the plugin to periodically send the logs to the native part, because if my app is heavy on native code it's nice to have the flutter logs mingled together with native logs in case my native code crashes... even if not in real time.. i understand that this is not the core use case of this plugin, but i also don't think providing this `sendLogs()` method hurts the plugin either..